cmd/spire-agent/cli/run/run.go

```diff
@@ -423,6 +426,24 @@ func NewAgentConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool)
 
 	ac.AllowUnauthenticatedVerifiers = c.Agent.AllowUnauthenticatedVerifiers
 
+	for _, authorizedDelegate := range c.Agent.AuthorizedDelegates {
+		_, err := common_cli.ParseTrustDomain(authorizedDelegate, logger)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
It doesn't seem that this is required, looks like the checks below are all what we need.

Reply:
hey @amartinezfayo, I used `ParseTrustDomain()` validation because its end up calling the functions to parse the SPIFFE ID - to see if the SPIFFE ID is compliant. For example, if I use these two SPIFFE ID as `authorized_delegates`, it shows the errors in SPIFFE ID format:
```
could not parse trust_domain "spiffe://example.org/#/aa": spiffeid: fragment is not allowed
```
```
could not parse trust_domain "spiffe://example.org/?/aa": spiffeid: query is not allowed
```

In order to avoid user to put a non-compliant SPIFFE ID.. do you think that it is not necessary?

Reply:
I see, thanks for pointing that!
I would use a single check, like:
```
		if err := idutil.ValidateSpiffeID(authorizedDelegate, idutil.AllowTrustDomainWorkload(ac.TrustDomain)); err != nil {
			return nil, fmt.Errorf("error validating authorized delegate: %v", err)
		}
```
That check would cover the validation of the SPIFFE ID and also checking the matching with the trust domain.

Reply:
Perfect!! Just test it and look very good option :D

tkss